code/datums/craft/gun_parts.dm

```diff
@@ -219,16 +219,16 @@
 		log_debug("legal_mechanism refence == [legal_mechanism]")
 		log_debug("legal_barrel refence == [legal_barrel]")
 
-		if(legal_grip != frame_to_grip)
-			to_chat(user, SPAN_WARNING("\the [src] dosnt have a proper grip for the componence's!"))
+		if(legal_grip != frame_to_grip) // ATF check
+			to_chat(user, SPAN_WARNING("The grip installed on \the [src] is too cheap to assemble this kind of high-grade gun!"))
```

Comment:
```suggestion
			to_chat(user, SPAN_WARNING("\the [src] dosnt have a proper grip for the componence's"))
```

Reply:
In no way am I ever reverting to an absolutely intelligible garbling that I cannot understand what it means

The way I worded it is intuitive as to let players know that "cheap 6.5mm barrel" is not gonna cut it if they want to assemble a Mamba, they'll need an actual proper one, which is what this check does as doccumented on its code commentary
Using "proper" doesn't tell me anything. I slapped a 6.5mm barrel and the frame accepted it, so why can't I assemble the gun? Saying that it's "cheap" will make people wonder "If this is cheap, then which is the actual good one?" until they either find one, or consult marshals for one
"Compnence" is a non-existant word which I assume means "components", which again neither does it tell me anything about why I cannot assemble the gun, and that "component" and "mechanism" can be easily missunderstood by non-english natives, not to mention that's not what's wrong on this case of assembly to throw this error message.